vm/interpreter/src/default_runtime.rs

```diff
@@ -55,9 +58,15 @@ struct VMMsg {
 
 impl MessageInfo for VMMsg {
     fn caller(&self) -> &Address {
+        assert!(
+            matches!(self.caller.protocol(), Protocol::ID),
+            "runtime message caller was not resolved to ID address"
+        );
         &self.caller
     }
     fn receiver(&self) -> &Address {
+        // * Can't assert that receiver is an ID address here because it was not being done
+        // * pre NetworkVersion3. Can maybe add in assertion later
```

Comment:
If this is part of the v2 actors and syncing with mainnet, shouldn't you just get that in now? Or is it a big effort that should be done in another PR?

Reply:
No like we can't do it at all, because it breaks in <v3. There is no possible way to do this without not supporting those versions

Reply:
it's one assertion